tools/generator/SourceWriters/BoundMethodAbstractDeclaration.cs

```diff
@@ -52,6 +52,11 @@ public BoundMethodAbstractDeclaration (GenBase gen, Method method, CodeGeneratio
 			if (method.DeclaringType.IsGeneratable)
 				Comments.Add ($"// Metadata.xml XPath method reference: path=\"{method.GetMetadataXPathReference (method.DeclaringType)}\"");
 
+			// TODO: shouldn't `[Obsolete]` be added for *all* CodeGenerationTargets?
```

Comment:
Seems like it would make sense to always prevent that warning. In the real world it's nearly impossible to bind libraries with `<TreatWarningsAsErrors>`, but it's something we should continue working towards.

Reply:
@jpobst: The implicit question here is whether *all* CodeGenerationTargets should add `[Obsolete]` to abstract methods, or *just* `CodeGenerationTarget.JavaInterop1`.

I opted to only update `CodeGenerationTarget.JavaInterop1` for expediency, but I suspect this should be done for everything.